tests/utils.py

```diff
@@ -146,3 +148,64 @@ def error_occurs_only_once(error_message: str, full_error: str) -> bool:
         return False
 
     return full_error.count(error_message) == 1
+
+
+@contextmanager
+def create_temp_file(content: str, filename: str) -> Generator[Path, None, None]:
+    """Create a file in a temporary directory with the specified file name and the provided content.
+
+    Args:
+        content (str): The content of the file as string.
+        filename (str): The file name.
+
+    Example:
+        >>>     with create_temp_file("some_file_content", "some_file_name") as temp_file:
+        >>>         print(f"Do something with {temp_file}")
+
+    Returns:
+        Generator[Path, None, None]: Generator with the path to the file in the temporary directory as yield type.
+    """
+    with get_temp_file(filename) as file:
+        with open(file, "w") as f:
+            f.write(content)
+        yield file
+
+
+@contextmanager
+def create_temp_file_from_lines(
+    lines: List[str], filename: str
+) -> Generator[Path, None, None]:
+    """Create a file in a temporary directory with the specified file name and the provided content.
+
+    Args:
+        content (str): The content of the file as list of string (lines of the file).
+        filename (str): The file name.
+
+    Example:
+        >>>     with create_temp_file_from_lines(["some_file_content"], "some_file_name") as temp_file:
+        >>>         print(f"Do something with {temp_file}")
+
+    Returns:
+        Generator[Path, None, None]: Generator with the path to the file in the temporary directory as yield type.
+    """
+    content = "\n".join(lines)
+    with create_temp_file(content, filename) as file:
+        yield file
+
+
+@contextmanager
+def get_temp_file(filename: str) -> Generator[Path, None, None]:
```

Comment:
Can you not do without this function, and on the (single) call site in `create_temp_file(...)`, use `tempfile.NamedTemporaryFile(...)` instead?

Reply:
ok to postpone, but I don't think a wrapper class would be needed. It would be a one-liner change on the call site, if I'm seeing this correctly.